internal/services/oracle/db_nodes_data_source_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright © 2024, Oracle and/or its affiliates. All rights reserved
+
+package oracle_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+)
+
+type DBNodesDataSource struct{}
+
+func TestDBNodesDataSource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_oracle_db_nodes", "test")
+	r := DBNodesDataSource{}
+
+	data.DataSourceTest(t, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("db_nodes.0.cpu_core_count").Exists(),
+			),
+		},
+	})
+}
+
+func (d DBNodesDataSource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+%s
+
+data "azurerm_oracle_db_nodes" "test" {
+  resource_group_name   = azurerm_resource_group.test.name
+  cloud_vm_cluster_name = azurerm_oracle_cloud_vm_cluster.test.name
+}
+`, d.template(data))
+}
+
+func (d DBNodesDataSource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+`, CloudVmClusterResource{}.basic(data))
+}
```

Comment:
There's no need to define the Cloud VM Cluster resource in a template 
```suggestion
func (d DBNodesDataSource) basic(data acceptance.TestData) string {
	return fmt.Sprintf(`
%s

data "azurerm_oracle_db_nodes" "test" {
  resource_group_name   = azurerm_resource_group.test.name
  cloud_vm_cluster_name = azurerm_oracle_cloud_vm_cluster.test.name
}
`, CloudVmClusterResource{}.basic(data))
}
```